source/server/connection_handler_impl.cc

```diff
@@ -264,6 +264,9 @@ void ConnectionHandlerImpl::ActiveTcpSocket::newConnection() {
       socket_->setDetectedTransportProtocol(
           Extensions::TransportSockets::TransportProtocolNames::get().RawBuffer);
     }
+    // Erase accept filter states because accept filters may not get the opportunity to clean up.
```

Comment:
Can you add a TODO here and file an issue around figuring out a more structural way of preventing this from happening? I think as we move towards IoHandles being produced by the dispatcher, it should be possible internally to figure out this case.

Reply:
will do

Reply:
Is it possible to add a quick unit test that at least makes sure that an accept filter is destructed before the connection is created? I think that shouldn't be too hard? @lizan WDYT?

Reply:
It's super hard to link everything together.
I tried to to verify accept_filters empty but it requires real world internal structs ActiveTcpXXX
I also tried verify the sequence of operations of FileEvent but it also need some surgery.

@lizan  Any advice?

Reply:
I have a super simple solution. Hold on

Reply:
updated